src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/StatusAndTiming.java

```diff
@@ -301,9 +302,25 @@ public static GenericStatus computeChunkStatus2(@Nonnull WorkflowRun run,
                 return GenericStatus.FAILURE;
             }
         }
+        WarningAction warning = findWarningBetween(firstNode, lastNode);
+        if (warning != null) {
+            return GenericStatus.UNSTABLE;
+        }
 
         // Previous chunk before end. If flow continued beyond this, it didn't fail.
-        return (run.getResult() == Result.UNSTABLE) ? GenericStatus.UNSTABLE : GenericStatus.SUCCESS;
+        return GenericStatus.SUCCESS;
+    }
+
+    private static @CheckForNull WarningAction findWarningBetween(@Nonnull FlowNode start, @Nonnull FlowNode end) {
```

Comment:
Maybe it would make more sense to move this method into `WarningAction.java` so that it could be used in other places like [FlowNode.getIconColor](https://github.com/jenkinsci/workflow-api-plugin/blob/c309efccb9492fd32fa4ac231b62a595a6913dcd/src/main/java/org/jenkinsci/plugins/workflow/graph/FlowNode.java#L282) (would only matter for block-scope stages, for atomic steps like `junit` just `getPersistentAction(WarningAction.class)` would suffice).